Orange/preprocess/preprocess.py

```diff
@@ -568,3 +568,52 @@ def __call__(self, data):
         for pp in self.preprocessors:
             data = pp(data)
         return data
+
+
+class SmartNormalize(Preprocess):
+    """
+    Construct a preprocessors that normalizes or merely scales the data.
+    If the input is sparse, data is only scaled, to avoid turning it to
+    dense. Parameters are diveded to those passed to Normalize or Scale
+    class. For more details, check those classes.
+
+    Parameters
+    ----------
+
+    zero_based : bool (default=True)
+        passed to Normalize
+
+    norm_type : NormTypes (default: Normalize.NormalizeBySD)
+        passed to Normalize
+
+    transform_class : bool (default=False)
+        passed to Normalize
+
+    center :
+        passed to Normalize as boolean and to Scale as recieved in constructor
+
+    normalize_datetime : bool (default=False)
+        passed to Normalize
+
+    scale : ScaleTypes (default: Scale.NoCentering)
```

Comment:
Default seems to be Scale.Std but I'm not sure it makes sense to scale by std if you don't center the data at the mean (std is computed as deviation from the mean). Instead, default to Scale.Span?